### PR TITLE
removing out-dated assert on python API

### PR DESCRIPTION
**Pull request by jjsjann123**
removing out-dated asserts in python API `define_vector`;
adding a tests verifying the behavior

### CODE REVIEW

**Comment by jjsjann123** [comment]
!build

File: test/test_gpu3.cpp [hunk 1]
```diff
@@ -8664,6 +8664,31 @@ TEST_F(NVFuserTest, Reduction3DConstantIterationDomain) {
       executor_cache.fusion(), cg_outputs, inputs, {ref}, __LINE__, __FILE__);
 }
 
+TEST_F(NVFuserTest, FusionTensorRankLimit) {
```
Comment (wujingyue) [inline test/test_gpu3.cpp:8667]: Can we move this test to a different file? `test_gpu\d.cpp` are already enormous. It's better to only remove tests from them.

**Review by wujingyue — APPROVED** [review]

**Comment by jjsjann123** [comment]
!build

File: test/test_pointwise.cpp [hunk 2]
```diff
@@ -430,4 +430,29 @@ TEST_F(PointwiseTest, VIssue1567ectorizationFactorAnalysisCase3) {
   testValidate(fusion, cg_outputs, aten_inputs, __LINE__, __FILE__);
 }
 
+TEST_F(NVFuserTest, FusionTensorRankLimit) {
```
Comment (wujingyue) [inline test/test_pointwise.cpp:433]:
```suggestion
TEST_F(PointwiseTest, FusionTensorRankLimit) {
```

Also, `sum` doesn't seem pointwise. Maybe test_gpu_fused_reduction.cpp is a better home?

**Comment by jjsjann123** [comment]
!build

**Comment by jjsjann123** [comment]
!build

**Comment by jjsjann123** [comment]
Failing CIs aren't relevant. I'm merge this one as-is.

**Pull request merged**